pkg/networkservice/chains/nsmgr/vl3_test.go

```diff
@@ -213,3 +225,76 @@ func Test_vl3NSE_ConnectsTo_vl3NSE(t *testing.T) {
 	_, err = resolver.LookupIP(ctx, "ip4", "nsc1.vl3")
 	require.Error(t, err)
 }
+
+func Test_NSC_GetsVl3DnsAddressAfterRefresh(t *testing.T) {
+	t.Cleanup(func() { goleak.VerifyNone(t) })
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*15)
+	defer cancel()
+
+	domain := sandbox.NewBuilder(ctx, t).
+		SetNodesCount(1).
+		SetNSMgrProxySupplier(nil).
+		SetRegistryProxySupplier(nil).
+		Build()
+
+	nsRegistryClient := domain.NewNSRegistryClient(ctx, sandbox.GenerateTestToken)
+
+	nsReg, err := nsRegistryClient.Register(ctx, defaultRegistryService("vl3"))
+	require.NoError(t, err)
+
+	nseReg := defaultRegistryEndpoint(nsReg.Name)
+
+	var serverPrefixCh = make(chan *ipam.PrefixResponse, 1)
+	defer close(serverPrefixCh)
+
+	serverPrefixCh <- &ipam.PrefixResponse{Prefix: "10.0.0.1/24"}
+	dnsServerIPCh := make(chan net.IP, 1)
+
+	_ = domain.Nodes[0].NewEndpoint(
+		ctx,
+		nseReg,
+		sandbox.GenerateTestToken,
+		vl3.NewServer(ctx, serverPrefixCh),
+		vl3dns.NewServer(ctx,
+			dnsServerIPCh,
+			vl3dns.WithDomainSchemes("{{ index .Labels \"podName\" }}.{{ .NetworkService }}."),
+			vl3dns.WithDNSPort(40053)),
+	)
+
+	resolver := net.Resolver{
+		PreferGo: true,
+		Dial: func(ctx context.Context, network, address string) (net.Conn, error) {
+			var dialer net.Dialer
+			return dialer.DialContext(ctx, network, "127.0.0.1:40053")
+		},
+	}
+
+	nsc := domain.Nodes[0].NewClient(ctx, sandbox.GenerateTestToken)
+
+	reqCtx, reqClose := context.WithTimeout(ctx, time.Second*10)
+	defer reqClose()
+
+	req := defaultRequest(nsReg.Name)
+	req.Connection.Labels["podName"] = nscName
+
+	resp, err := nsc.Request(reqCtx, req)
+	require.NoError(t, err)
+	req.Connection = resp.Clone()
+	require.Len(t, resp.GetContext().GetDnsContext().GetConfigs(), 0)
+
+	dnsServerIPCh <- net.ParseIP("127.0.0.1")
+
+	respAfterRefresh, err := nsc.Request(reqCtx, req)
+	require.NoError(t, err)
+	require.Len(t, respAfterRefresh.GetContext().GetDnsContext().GetConfigs(), 1)
+	require.Len(t, respAfterRefresh.GetContext().GetDnsContext().GetConfigs()[0].DnsServerIps, 1)
```

Comment:
Note: We don't need to do a refresh request manually here. 

nsc should receive the event from the nse. And right after that `refresh` chain element should call the request.

So, add an injectconnection chain element for the nsc chain and make sure that refresh for DNS is happening.